### ISSUE TITLE
Difference in rendering with native renderer between versions

### ISSUE THREAD

**Issue by jacovandermolen** [trevorsandy/lpub3d#731]
### Subject
Difference in rendering with native renderer between versions

### Environment
* Versions of LPub3D
* 2.4.7.r2
* 2.4.5.r62
* 2.4.5.r144

There is a difference in rendering with native renderer between versions, same settings and same PC.
I'd like to see the latest versions render like the 2.4.5x versions?
For reference see these images.

![31134 - Space Shuttle_150_DPI_page_1_2 4 7r2](https://github.com/trevorsandy/lpub3d/assets/32261647/f67195b6-2ac5-4f6a-91c6-5493611c9754)
![31134 - Space Shuttle_150_DPI_page_1_2 4 5r62](https://github.com/trevorsandy/lpub3d/assets/32261647/4eff89da-5e99-4c8e-b83d-890141e240d5)
![31134 - Space Shuttle_150_DPI_page_1_2 4 5r144](https://github.com/trevorsandy/lpub3d/assets/32261647/1931b247-9802-47b7-8e41-e6f7428c0a83)

**Comment by trevorsandy** [comment]
> For reference see these images.

Was the first image taken from 2.4.7.r2, the second from 2.4.5.r62 and the third from 2.4.5.r144 ?

What is the main difference you are seeing ? Is it the model scale, lines or level of detail ?

Cheers,

**Comment by jacovandermolen** [comment]
Yes, sorry, in that order.
The main difference is the level of detail I guess.
2.4.7 is somewhat blurry, more anti-aliased?
2.4.5 lines are sharp. Line on black part is white, where the outline in 2.4.7 on black part is more light grey.
There is a scale difference between 2.4.5.62 and 144 and shape of PLI differs.
Why is that?

**Comment by trevorsandy** [comment]
Thank you for the details.

For comparison, can you share the simple model file (and header file if applicable) used to generate the images above ?

Cheers,

**Comment by jacovandermolen** [comment]
Sure, here it is.
FYI: I am busy doing some demos again regarding LDraw: modelling with LDCad and then making instructions with LPub3D.
WIP on the 31134 3-in-1 creator set
[31134 - Space Shuttle.mpd.txt](https://github.com/trevorsandy/lpub3d/files/11880783/31134.-.Space.Shuttle.mpd.txt)
[header.ldr.txt](https://github.com/trevorsandy/lpub3d/files/11880784/header.ldr.txt)

I am also curious how this renders with the latest releases of LPub3D.
I am getting some errors here and there.

**Comment by jacovandermolen** [comment]
Here are the "helpers" for the arrows
[Helpers.zip](https://github.com/trevorsandy/lpub3d/files/11881038/Helpers.zip)
put in C:\Users\Public\Documents\LDraw\Unofficial or similar.

**Comment by trevorsandy** [comment]
Excellent. Many thanks.

**Comment by jacovandermolen** [comment]
Have you found a way to correct this? I want to try the new versions but if they render like they do now, I cannot work with it.

**Comment by trevorsandy** [comment]
Oops, I completely overlooked this ticket. 

I'll take a look.

Cheers,

**Comment by jacovandermolen** [comment]
Tnx!

**Comment by trevorsandy** [comment]
If you are running a portable distribution, can you send a copy of the LPub3D.ini so I can compare your settings with mine ? Thanks.

Cheers,

**Comment by jacovandermolen** [comment]
I found out version 2.4.5.62.3270_20221207 renders correct, atleast the way I want with the native rendere both perspective as orthographic. This is its ini.
[LPub3D - kopie.ini.txt](https://github.com/trevorsandy/lpub3d/files/12515823/LPub3D.-.kopie.ini.txt)

**Comment by jacovandermolen** [comment]
OK, something strange is going on with the PC that produces the other rendering.
My laptop at home now renders like I want in both ortho and perspective mode with the native renderer using the latest build 2.4.7.r72.

So only you have the slightest idea what is causing the behavior, you can close this issue.
Seems related to my PC at work.

**Comment by trevorsandy** [comment]
> So only you have the slightest idea what is causing the behavior, you can close this issue.
Seems related to my PC at work.

Very well. 

> There is a scale difference between 2.4.5.62 and 144 and shape of PLI differs.
Why is that?

At some point after 2.5, I changed the camera default distance factor - I can't remember exactly, I'll have to look up the commits to tell exactly at what rev. 

An editor can set/change the default 'factor' to align the size of rendered images between the Native renderer and LDView/LDGLite. If you can check this setting (3DViewer Preferences) on 2.4.5.r62 and it is different from 2.4.7.r72, this will likely be the reason for the size change. 

Further, the rendered image size change will affect the calculated width and height of the PLI background; hence, the difference there.

You can see more details in the 'What's this' (?) help link at the top right of the dialog - See #618.

![Default_Size_01](https://github.com/trevorsandy/lpub3d/assets/13388970/ede7ed70-fbe5-4270-8fe0-b596275cde98)

Cheers,

**Comment by jacovandermolen** [comment]
OK, I see. The size difference I can live with ;-)

The one thing I do not understand is the way the outlines are rendered. Anything on that?

**Comment by trevorsandy** [comment]
> The one thing I do not understand is the way the outlines are rendered. Anything on that?

Under Global Project Setup:
![Outlines_01](https://github.com/trevorsandy/lpub3d/assets/13388970/11c52d24-c2a3-4ebe-8ba0-e57fa07f947e)

Stud Style and Automate Edge Color:

![Outlines_02](https://github.com/trevorsandy/lpub3d/assets/13388970/fe927e04-8d64-48c4-a84c-b1a5216075e4)

Have you looked at Automate edge colors ?

![Outlines_03_Automate_Edge_Colours](https://github.com/trevorsandy/lpub3d/assets/13388970/d14e6f5a-2b0e-4470-bf66-31b3691a16c8)

And High-contrast stud style color settings ?

![Outlines_04_HC_Edge_Colours](https://github.com/trevorsandy/lpub3d/assets/13388970/b3acda37-b3aa-4b08-8276-2eab3a77fd43)

Cheers,

**Comment by jacovandermolen** [comment]
Yes, all of those settings are set. They are included in the header.ldr file I use for most of my models.
Left is 2.4.5 r26 Right 2.4.7 r72 Same settings, same computer
![main-0_1_0_0_0_1_0_0_0_1_1_1753_150_DPI_1 3_30_37_55_0_180_0_REL](https://github.com/trevorsandy/lpub3d/assets/32261647/b7a4bfcd-24f5-4dea-944f-e854b6ecadf5)![main-0_1_0_0_0_1_0_0_0_1_1_1753_150_DPI_1 3_30_37_55_0_180_0_REL](https://github.com/trevorsandy/lpub3d/assets/32261647/94c6c883-7352-4f45-930b-f911e2d9df00)

**Comment by trevorsandy** [comment]
> The one thing I do not understand is the way the outlines are rendered. Anything on that?

No. For me, there is no edge line difference between v2.4.5r26 and v2.4.7 latest.

v2.4.5r26
![731_v2 4 5r26](https://github.com/trevorsandy/lpub3d/assets/13388970/13090e1c-49ab-4ee1-ba83-0f2b5186bdb6)

v2.4.7r72
![731_v2 4 7r72](https://github.com/trevorsandy/lpub3d/assets/13388970/582b9f6c-29cb-4ff4-9365-3832b9f5080f)

On the other hand, I was able to identify the reason for the difference in rendered image size. There is an arbitrary FOV adjustment for Native rendered images that was exposed in the `LPub3D.ini` file as `NativeImageCameraFoVAdjust` with a default value of _6_. In v2.4.6, this adjustment was changed to _15_ (hence the smaller images) and set as a macro so it is no longer exposed in the `LPub3D.ini`. If you wish to adjust the rendered size of your images, use the `CAMERA_FOV` meta command.

The v2.4.7r72 image above is achieved with `0 !LPUB ASSEM CAMERA_FOV 20` (the default FOV for the Native renderer is _30_).

In the next release, I will restore `NativeImageCameraFoVAdjust` to the `LPub3D.ini`, but the default value will remain _15_. You will only be able to directly change the `LPub3D.ini` setting (e.g. back to _6_) if you so desire - no UI will be available.

Cheers,
